### PR TITLE
Robot artifacts

### PR DESCRIPTION
**Pull request by amylizzle**
## About the PR 
Adds a new artifact type: Drones

Intended for the clockwork artifacts (#21375), but also ancient origin.

When activated, the artifacts become robot mobs that do one of a few things:

- wandering
- smashing walls
- replacing floors with a randomly chosen type
- placing walls on empty floors (not space)
- attacking people

## Why's this needed? 
It's a base for an artifact type that fits with the clockwork theme

## Changelog 

```changelog
(u)Amylizzle
(*)Adds a new artifact type: Drones
```

### CODE REVIEW

File: code/mob/living/critter/ai/artifact_robot.dm [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+	max_dist = 7
+
+/datum/aiTask/sequence/goalbased/wall_smash/New(parentHolder, transTask) //goalbased aitasks have an inherent movement component
+	..(parentHolder, transTask)
+	add_task(holder.get_instance(/datum/aiTask/succeedable/wall_smash, list(holder)))
+
+/datum/aiTask/sequence/goalbased/wall_smash/get_targets()
+	. = ..()
+	var/list/turf/walls = list()
+	for(var/turf/simulated/wall/W in view(src.max_dist, src.holder.owner))
```
Comment (TobleroneSwordfish) [inline code/mob/living/critter/ai/artifact_robot.dm:27]: Make it target girders too so it does actual deconstruction?

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
Woe, `New` argument inheritance: 
image

Ordering issue on death gib:
image

Maybe the robot mobs should have brullbar-like stamina stun immunity, it looks weird to see an artifact get stamina crit with a chair.

Not sure if this an artifact of the init runtimes but they currently seem to pick a new AI each time they're activated/deactivated, ideally I think that should be persistent.

File: code/obj/artifacts/artifact_machines/robot.dm [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	validtypes = list("ancient")
+	fault_blacklist = list(ITEM_ONLY_FAULTS,TOUCH_ONLY_FAULTS)
+	activ_text = "whirs to life!"
+	deact_text = "becomes eerily still."
+	// possible AI types that the robot can have
+	react_xray = list(50,20,90,8,"MECHANICAL")
+	var/static/list/datum/aiHolder/possible_ais = list(/datum/aiHolder/artifact_wallplacer, /datum/aiHolder/artifact_wallsmasher, /datum/aiHolder/artifact_floorplacer, /datum/aiHolder/wanderer, /datum/aiHolder/aggressive)
+	// possible floor types for the floor placing robots
+	// possible wall types for the wall placing robots
+	var/static/list/turf/floor_types = list(/turf/simulated/floor/industrial, /turf/simulated/floor/mauxite, /turf/simulated/floor/circuit/vintage, /turf/simulated/floor/glassblock/transparent, /turf/simulated/floor/void, /turf/simulated/floor/techfloor/yellow)
```
[the remark this answers is not in the document]
Comment (amylizzle) [reply code/obj/artifacts/artifact_machines/robot.dm:20]: huh it does not.

additional opendream bug discovered :laughing:

**Comment by amylizzle** [comment]
Ah it was supposed to be persistent ai types but I accidentally left some debug code in place. Fixed now.

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
The actionbars and sounds are great, I love watching them fight each other over walls.
- The mauxite floors don't seem to be changing the floor texture? I swear they were before but 🤷 
- New runtime:
image
Occurs on artifact activation, seems like the `huds` list is an instance.

**Pull request merged**